<?phpclass ExceptionHook{	public function SetExceptionHandler()	{		set_exception_handler(array($this, 'HandleExceptions'));	}		public function HandleExceptions($exception)	{    $DEBUG=true;		$msg ='Exception of type \''.get_class($exception).'\' occurred with Message: '.$exception->getMessage().' in File '.$exception->getFile().' at Line '.$exception->getLine();		$msg .="\r\n Backtrace \r\n";		$msg .=$exception->getTraceAsString();		$CI = &get_instance();   					if($CI->config->item('DEBUG'))		{			$CI->load->library('EmailLibrary');						$CI->emaillibrary->inizialize(array('protocol' => 'mail'));      $CI->emaillibrary->send(array('mail_from' => 'debug@jlbbooks.it', 'mail_from_name' => 'DEBUG', 'mail_to' => $CI->config->item('EMAIL_DEBUG'), 'subject' => 'PHP EXCEPTION FROM: '.base_url(), 'message' => $msg));						}		show_error($msg);		}		} 